packages/stylelint-plugin/src/rules/no-focus-visible/no-focus-visible.ts

```diff
@@ -0,0 +1,72 @@
+import * as stylelint from 'stylelint';
+import * as parsel from 'parsel-js';
+import { extractAllSelectors } from '../../postcss-utils';
+import { createRule } from '../../create-rule';
+import { normalizeRuleName } from '../../normalize-rule-name';
+
+const ruleName = normalizeRuleName('no-focus-visible');
+
+const messages = stylelint.utils.ruleMessages(ruleName, {
+  focusVisible: (selector: string) =>
+    `Selector ${selector} contains :focus-visible`,
+  focus: (selector: string) => `Selector ${selector} contains :focus`,
+  parseError: (selector: string) => `Failed to parse selector ${selector}`,
+});
+
+export default createRule({
+  messages,
+  ruleName,
+  ruleFunction: () => (postcssRoot, postcssResult) => {
+    const validOptions = stylelint.utils.validateOptions(
+      postcssResult,
+      ruleName
+    );
+
+    if (!validOptions) {
+      return;
+    }
+
+    const { selectorToRule, selectors } = extractAllSelectors(postcssRoot);
+
+    for (const selector of selectors) {
+      let tokenizedSelector = [];
+      try {
+        tokenizedSelector = parsel.tokenize(selector);
```

Comment:
Do you really need to tokenize there? Can't we simply use `.indexOf(":focus-visible")`?